### DISCUSSION TITLE
Max Connections?

### DISCUSSION THREAD

**Discussion by cecom** [screego/server#117] [General]
Hi,

is there a limit how many can connect at the same time in one room with one person presenting something? We are using STUN. We tried it today with 20+, but then we got stability problems.

**Comment by jmattheis** [comment]
Hey @cecom,

there is no internal limit for connections, but it is limited by the upload rate of the user that shares the screen. If you have 20 users viewing the stream, then the user that shares the screen has to upload the screen to 20 different users. As you can see, this doesn't scale really well.

This could be fixed in screego by f.ex letting screego acting as relay (not meaning using TURN), then the user that shares only have to upload the stream once, and the server will forward the stream to each viewer. This has some downsides, because then the video stream won't be end-to-end encrypted between the users anymore and the server can intercept the stream much more easily.

There are other ways, where each viewer may restream the video to other viewers, but this is fairly complex.

I try to keep screego as simple as possible, and it currently completely satisfies my use-cases, thus, these both possible fixes are out of scope for this project. If you need something like this, then you can have a look at Jitsi. If there are many users, then Jitsi should do stuff like my first suggestion for fixing this issue.

**Reply by cecom** [reply]
Hey @jmattheis, thx for your answer. Simple is better, so great work!
